### ISSUE TITLE
What do you most want to see improved in Social Engine?

### ISSUE THREAD

**Issue by Elshara** [SocialEngine/phpv4-issues#332]
We've all contributed a great deal to the improvement of the product so far. Some more popular feedback topics have left us wondering, not if it will be implemented, not even when it will be released, but rather how will it improve our websites when it is built.
I can relate to how you're feeling. Getting our hands on new release code just isn't possible with the current way social engine updates its code. So, I thought I'd start a thread on what improvements you're looking forward using in a practical sense, regardless of whether it is built, upcoming or on the drawing board.
Here are my top 10.
1. A better caching system. Simply put, static content should be cached where as dynamic content should be instant. Better control over what gets cached may be the best thing we have to a practical performance improvement.
2. PHP7 support. If wordpress can update to responsive themes and already has php7 implemented, social engine with its meriad of fans and supporters should consider redefining the code structure to work with modern browsers and modern improvements the internet has to offer. If the developers don't consider this a priority, perhaps wordpress importing of social engine plugins as a free tool would change their minds. We invest readily into this company, we should receive code improvements, not bug fixes.
3. A better way to handle post editing. Simply put, tags need to go. They are rarely used on social engine, and if they are to be used, I think repositioning the word tag to be "subjects covered" or "related phrases" would be a much better fit. In fact, replacing the tag system with a box to have perhaps a post signature or prologue and adding a summary box to go along with it, would definitely be up to industry standard. In addition, replacing all flash requirements with html5 will ensure services like plupload or uploadify bring us to the forefront of modern developments. Did you know: Facebook has already moved on from php and mysql years ago? If we want to have a modern social network, we ought to look into that as well. It starts with how people post things, if it is modern, others will enjoy it better and it would run faster as well.
4. A better way to handle plugins. We should have most of our page content as post html and compress all other files to one css and javascript file. This way it can reference what it needs and we can do away with things like mootools and other various plugin core requirements so the website looks and loads much faster, cleaner with out all the extension requirements. I mean really, having to not even be able to put spaces in your username, then having to look at profiles that can't even determine one name space from another, isn't that 1990's technology? Doing away with content id and database reliance for a start will definitely take us in the right direction, and one step forward to a modern website. If wordpress can do it, so can we.
5. A better media player. Currently, you have to upload media to three basic modules. Photos, music and videos, rely on separate codecs to handle their respective files. What if however, they could be merged into one, have mass upload enabled through html5, had browsers implement the ability to play the files and download its metadata through javascript and be able to divide files not by file type extension, but by audio, video and photos. A media gallery plugin would take care of this.
6. Merge the polls and forums plugin into one. Polls should be apart of a forum discussion, this is how most modern forum boards do things. Simply put, all user engagement should run through one idea, and having polls as a forum attachment would make this a lot easier.
7. Merge the groups and events plugin into one. All events do is create something that will only be used once. Who needs an event profile when it won't be used regularly. With events as an attachment to a group post, all of this would become clear since you'd have to join a group anyway for it to become an event. Events are bigger groups, so it makes sense to make it apart of the groups plugin.
8. Merge the marketplace plugin into an attachment. This way paid access could be implemented by both admins and members. Allowing someone to advertise in one central location, isn't convenient for people anymore because if they don't find what they're looking for, they'll simply not use it. Having the marketplace as an attachment allows the selling and receiving of an item to be tightly integrated into what ever page someone is currently viewing. It works both as a paid access system with multiple contributors, and also as a sails management system since its primary focus is to get money from one person to another. Attaching it to all forms of posts with its current set up would make this a lot easier for people to sell something and still get their point across if they are interested.
9. Make activity feeds a thing of the past. To be replaced with post specific comments by paragraph. Soundcloud is already doing this and it makes sense because while you're reading a post, you can in parallel, read its relative comments based on the current position. This can be achieved using slideshow and ajax based technology. Activity feeds record items separately, however apart from the main and profile activity feed, if it was integrated into posts where members can click and drag sections to like, dislike and comment on, it would be a lot easier for people to engage and notify each other of where things make the most sense and why they are important.
10. Update email templates on a global scale. I am so tired of social engine using unprofessional emails to communicate activity with the recipients. I mean really, who puts social network administration at the bottom of all emails without a subject line. The default email address is "email@domain.com" why bother filling in that field, why not make it apart of the set up process for installing social engine. While I'm at it, just use the site name and language templates to automatically be supported in all email templates in future so members get a lot more detailed emails for their activity. Why should we have to hand code emails just so that it matches what can be supported or is already. And please, include an unsubscribe link at the bottom of all emails and update all email phrases to a global signature.
Feel free to add to this list or make your own out of it. I thought this would help make a clear and concise point across.

**Comment by eugene-wh** [comment]
Excellent points.

As to caching our team has actually tried to release a plugin offering more advanced caching (so static pages are saved and mysql popular queries are cached). But we have found that close to impossible to get done within plugin limitations.

That should definitely be a part of platform itself.

**Comment by meetgerry** [comment]
I second Eugene!

**Comment by Elshara** [comment]
I third you.

**Comment by Elshara** [comment]
Perhaps it may be possible for all mysql queries to be cached as in
cache or update current database tables via cron job then if a new one
has to be created, it can simply do so in real time independent of
what else has been cached so whatever is cached will be served first
so long as it is static, which can be everything in and of itself.

**Comment by smazurov** [comment]
Query Cache is something you can implement transparently and independently of SE or PHP. See [this](https://dev.mysql.com/doc/refman/5.7/en/query-cache.html) as a jump off point.

**Comment by Elshara** [comment]
Awesome!

**Comment by gsf00001** [comment]
Many good ideas ...

#2 PHP7 support for sure - ASAP
A. Update to Zend Framework (and other foundational tools), especially due to security issues
B. User Quota - the ability to enter a value (GB with decimal) - the existing options are too few/limiting
C. Plugin & Licensing ideas (some here and in other posts)
D. Header/Footer - abililty from Layout Editor to enable/disable either or both on a per page basis

Not SE PHP related, but more overall:
- better communication from SE about upcoming releases and other things SE is working on as a company
- better review sections for Devs and Plugins, with different review options and questions per each

There are many more, but that's my short list.

**Comment by DonnaScriptTechs** [comment]
Many of these are improved for 4.9 when it releases. Please check at that time and please open one report per issue if you still see issues. For discussions such as this, please open a thread at the community.

**Comment by Elshara** [comment]
Awesome!
